packages/console/src/components/ImageInputs/index.tsx

```diff
@@ -0,0 +1,155 @@
+  type Control,
+  type UseFormRegister,
+  type FieldError,
+} from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import FormField from '@/ds-components/FormField';
+import Skeleton from '@/ds-components/FormField/Skeleton';
+import TextInput from '@/ds-components/TextInput';
+import ImageUploader from '@/ds-components/Uploader/ImageUploader';
+import useImageMimeTypes from '@/hooks/use-image-mime-types';
+import useUserAssetsService from '@/hooks/use-user-assets-service';
+import { uriValidator } from '@/utils/validator';
+
+import * as styles from './index.module.scss';
+
+export type ImageField<FormContext extends FieldValues> = {
+  /** The name (field path) of the field in the form. */
+  name: FieldPath<FormContext>;
+  /**
+   * The type of the field. It should match the existing structure in the translation file to get
+   * the correct translations.
+   */
+  type: keyof LocalePhrase['translation']['admin_console']['sign_in_exp']['branding_uploads'];
+  theme: Theme;
+  /** The error message of the field in the form. */
+  error?: FieldError;
+};
+
+type Props<FormContext extends FieldValues> = {
+  /** The condensed title when user assets service is available. */
+  readonly uploadTitle: React.ComponentProps<typeof FormField>['title'];
+  readonly control: Control<FormContext>;
+  readonly register: UseFormRegister<FormContext>;
+  readonly fields: Array<ImageField<FormContext>>;
+};
+
+/**
+ * A component that renders the logo inputs for a form.
+ *
+ * When user assets service is available, it will render the image uploader components side-by-side;
+ * otherwise, it will render the text inputs.
+ */
+function ImageInputs<FormContext extends FieldValues>({
+  uploadTitle,
+  control,
+  register,
+  fields,
+}: Props<FormContext>) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [uploadErrors, setUploadErrors] = useState<Partial<Record<string, string>>>({});
+  const { description } = useImageMimeTypes();
+  const { isReady: isUserAssetsServiceReady, isLoading } = useUserAssetsService();
+  const uploadErrorChangeHandlers = useMemo(
+    () =>
+      Object.fromEntries(
+        fields.map((field) => [
+          field.name,
+          (message?: string) => {
+            setUploadErrors((previous) => ({ ...previous, [field.name]: message }));
+          },
+        ])
+      ),
+    [fields]
+  );
+
+  if (isLoading) {
+    return <Skeleton formFieldCount={2} />;
+  }
+
+  if (!isUserAssetsServiceReady) {
+    return (
+      <>
+        {fields.map((field) => (
+          <FormField
+            key={field.name}
+            title={
+              <>
+                {t(`sign_in_exp.branding.with_${field.theme}`, {
+                  value: t(`sign_in_exp.branding_uploads.${field.type}.url`),
+                })}
+              </>
+            }
+          >
+            <TextInput
+              key={field.name}
+              {...register(field.name, {
```

Comment:
need to provide a fallback for empty value here, or the input will become "" on focused and the form will be dirty.

Reply:
This should be handled in the form default values?

Reply:
discussed offline